### PR TITLE
Added playlists page, single playlist page and api calls

### PR DESCRIPTION
**Pull request by yaswanthmyneni**
# Added playlists page, single playlist page, and API calls

## Preview: [lucent-video-lib](https://deploy-preview-4--ym-video-lib-dev.netlify.app/)
-----> Click on the categories on the homepage to visit the video listing page in the preview.
-----> On clicking the watch now button, it will be redirected to the video page and also adds the data to the history page at the same time.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -17,6 +20,7 @@ function App() {
     <>
       <Navigation />
       <Routes>
+        <Route path="*" element={<ErrorPage />} />
```
Comment (kgajriaAgarwal) [inline src/App.js:23]: suggestion : try creating a separate file for the routes for improving the code structure

**Review by kgajriaAgarwal — APPROVED** [review]
The code is nice , modular and clean.It can be merged.

File: src/components/video-card/VideoCard.jsx [hunk 2]
```diff
@@ -2,46 +2,90 @@ import {
...
+                onClick={
+                  watchLater
+                    ? () => {
+                        removeFromWatchLater();
+                        setDropdown(!dropdown);
+                      }
+                    : () => {
+                        addToWatchLater();
+                        setDropdown(!dropdown);
+                      }
```
Comment (sri-rishi) [inline src/components/video-card/VideoCard.jsx]: Suggestion: can extract function from here to another function and use , use it here

File: src/context/playlist-context.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        if (response.status === 200) {
+          playlistDispatch({
+            type: "PLAYLIST",
+            payload: response.data.playlists,
+          });
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    })();
```
Comment (sri-rishi) [inline src/context/playlist-context.js]: Suggestion: you can create separate file for api-calls and extract api call functions from there

**Review by sri-rishi — APPROVED** [review]
code and UI looks good, just some suggestions which I think can be done, feel free to merge

**Review by Sahil25061999 — APPROVED** [review]
code and ui looks good to me, not much suggestion from me, just a small suggestion for the ui, to have a button to navigate to video listing page.

**Pull request merged**